src/Psl/Filesystem/get_basename.php

```diff
@@ -14,17 +14,19 @@
  *
  * In other environments, it is the forward slash `/`.
  *
- * @param string|null $suffix If the filename ends in a suffix, this will also be cut off.
+ * @param non-empty-string|null $suffix If the filename ends in a suffix, this will also be cut off.
```

Comment:
Why suffix can not be an empty string?

Reply:
provide null in that case ^^ otherwise, it's useless ( this is done to encourage better typing down stream for users, so they know when they have a non-empty-string/empty-string. )

Reply:
Wouldn't `string $suffix = ''` be more convinient for users? `$maybeNull ?? ''` is easier to do than `$maybeEmpty === '' ? null : $maybeEmpty`.